test/golden/Test.hs

```diff
@@ -71,6 +71,13 @@ sumMon :: Monoid a => [a] -> a
 sumMon [] = mempty
 sumMon (x : xs) = x <> sumMon xs
 
+hd :: [a] -> a
+hd [] = undefined
```

Comment:
Interesting. Could we compile to `error` instead and get both totality and be more faithful to the Haskell code?

I guess one could have a custom eliminator for the empty type:
```
hd [] ⦃ () ⦄
```
```
hd [] ⦃ e ⦄ = error e "blah"
```

Reply:
Yes, absurd clauses should probably compile to `error "impossible"` rather than `undefined`.

Reply:
I think that would be nice and I guess even though uses of `hd` are guaranteed to be safe when writing Haskell code in Agda due to `NonEmpty` one can still use it unsafely from Haskell mode written directly in Haskell and trigger `error/undefined`.